Source/SteveSloth/Private/MyEnemyAttackingState.cpp

```diff
@@ -0,0 +1,69 @@
+/****************************************************************************************
+ * Copyright: SteveSloth
+ * Name: Ken Ferris
+ * Script: MyEnemyAttackingState.cpp
+ * Date: May 3, 2024
+ * Description: Enemy Attack State
+ * TODO:
+ * Known Bugs:
+ ****************************************************************************************/
+
+
+#include "MyEnemyAttackingState.h"
+#include "MyEnemyBaseClass.h"
+#include "Kismet/GameplayStatics.h"
+
+void UMyEnemyAttackingState::EnterState()
+{
+
+}
+
+void UMyEnemyAttackingState::ExitState()
+{
+
+}
+
+void UMyEnemyAttackingState::UpdateState(float deltaTime)
+{
+	if (IsPlayerInRange())
+	{
+		AttackPlayer();
+	}
+}
+
+bool UMyEnemyAttackingState::IsPlayerInRange() const
+{
+	AMyEnemyBaseClass* Owner = Cast<AMyEnemyBaseClass>(GetOuter());
+	if (!Owner)
+	{
+		UE_LOG(LogTemp, Error, TEXT("MyEnemyAttackingState::IsPlayerInRange - Owner is not valid"));
+		return false;
+	}
+
+	ACharacter* PlayerCharacter = UGameplayStatics::GetPlayerCharacter(GetWorld(), 0);
+	if (!PlayerCharacter)
+	{
+		UE_LOG(LogTemp, Error, TEXT("MyEnemyAttackingState::IsPlayerInRange - Player character not found"));
+		return false;
+	}
+
+	float Distance = FVector::Distance(Owner->GetActorLocation(), PlayerCharacter->GetActorLocation());
+	return Distance <= Owner->GetAttackRange();
+}
+
+void UMyEnemyAttackingState::AttackPlayer()
+{
```

Comment:
This Attack player does not attack the player... There should be combat in here, interaction with the player, playing animations, doing damage to the player. Things of that sort.